src/routes/transactions/transactions.service.ts

```diff
@@ -70,4 +73,54 @@ export class TransactionsService {
       results,
     };
   }
+
+  async getIncomingTransfers(
+    chainId: string,
+    routeUrl: Readonly<URL>,
+    safeAddress: string,
+    executionDateGte?: string,
+    executionDateLte?: string,
+    to?: string,
+    value?: string,
+    tokenAddress?: string,
+    paginationData?: PaginationData,
+  ): Promise<Partial<Page<IncomingTransfer>>> {
+    const transfers = await this.safeRepository.getIncomingTransfers(
+      chainId,
+      safeAddress,
+      executionDateGte,
+      executionDateLte,
+      to,
+      value,
+      tokenAddress,
+      paginationData?.limit,
+      paginationData?.offset,
+    );
+
+    const safeInfo = await this.safeRepository.getSafe(chainId, safeAddress);
+    const results = await Promise.all(
+      transfers.results.map(
+        async (transfer) =>
+          new IncomingTransfer(
+            await this.incomingTransferMapper.mapTransfer(
+              chainId,
+              transfer,
+              safeInfo,
+            ),
+          ),
+      ),
+    );
+
+    const nextURL = cursorUrlFromLimitAndOffset(routeUrl, transfers.next);
+    const previousURL = cursorUrlFromLimitAndOffset(
+      routeUrl,
+      transfers.previous,
+    );
+
+    return {
```

Comment:
For several routes we are omitting `next` and `previous` (when next/previous pages aren't needed and therefore their values are `undefined`). Probably for coherence we want to explicitly set them to `null`. Since this affects several routes also, I think we can create a separate issue as well, wdyt?

Reply:
Yes I agree. We should set them to `null` explicitly.

But we can also do it here too. `cursorUrlFromLimitAndOffset` already returns either a value or `null` so it's up to the route to map it `undefined` (which we shouldn't). No strong opinion on how you want to do this so I'll leave it up to you.

Reply:
Great, I'll fix this route pagination values here, and I've also created a separated PR to explicitly set these values to null when applicable into other routes 🙂 https://github.com/5afe/safe-client-gateway-nest/pull/208

Reply:
Great, I'll fix this route pagination values here, and I've also created a separated PR to explicitly set these values to null when applicable into other routes 🙂 https://github.com/5afe/safe-client-gateway-nest/pull/208

Reply:
Changed in [2f6d0c5](https://github.com/5afe/safe-client-gateway-nest/pull/206/commits/2f6d0c508eace06c134039e921fe51865fc195eb)